lib/opFns.js

```diff
@@ -16,99 +16,99 @@ module.exports = {
   ADD: function (a, b, runState) {
     return Buffer.from(
       new BN(a)
-        .add(new BN(b))
+        .iadd(new BN(b))
         .mod(utils.TWO_POW256)
         .toArray('be', 32))
   },
   MUL: function (a, b, runState) {
     return Buffer.from(
       new BN(a)
-        .mul(new BN(b))
+        .imul(new BN(b))
         .mod(utils.TWO_POW256)
         .toArray('be', 32)
     )
   },
   SUB: function (a, b, runState) {
     return utils.toUnsigned(
       new BN(a)
-        .sub(new BN(b))
+        .isub(new BN(b))
     )
   },
   DIV: function (a, b, runState) {
-    a = new BN(a)
     b = new BN(b)
 
     var r
     if (b.isZero()) {
       r = [0]
     } else {
+      a = new BN(a)
       r = a.div(b).toArray('be', 32)
```

Comment:
All this was intentional to keep the change diff small and reviewable. It is going to be replaced by #159 anyway.